src/panels/config/helpers/ha-config-helpers.ts

```diff
@@ -243,6 +256,14 @@ export class HaConfigHelpers extends SubscribeMixin(LitElement) {
             } else if (message.type === "updated") {
               newEntries[message.entry.entry_id] = message.entry;
             }
+            if (
+              this._entitySource &&
+              this._configEntries &&
+              message.entry.state === "loaded" &&
+              this._configEntries[message.entry.entry_id]?.state !== "loaded"
+            ) {
+              this._debouncedFetchEntitySources();
```

Comment:
I needed to add some delay here because I think this is racy, if I just refetch sources here immediately after state changes to "loaded" about 10% of the time the newly loaded entity would not be present in the sources list returned from core.